### DISCUSSION TITLE
Asset tags auto-increment

### DISCUSSION THREAD

**Discussion by Viiviv** [snipe/snipe-it#9507] [Q&A] [answered]
Hi,

I have an issue with auto-increment with the asset tags.
The auto-increment setting doesn't appear to be working correctly.
If I try to set the increment in the menu ( https://snipe-it.readme.io/docs/asset-tags )
It doesn't change in the UI. It does however change in the database settings next_auto_tag_base
Why could this be happening?

Using Version v5.1.4 - build 5886 (master)

**Comment by slohse** [comment, marked as answer]
I had the same problem. It ended up being some kind of divergence between the cache and the database.
As I use the `file` CACHE_DRIVER, I moved the `storage/framework/cache` elsewhere. I had to re-configure the asset tag settings afterwards, but now it seems to work again :crossed_fingers:

**Reply by slohse** [reply]
Also, while digging I found this:

```
    public static function autoincrement_asset()
    {
        $settings = \App\Models\Setting::getSettings();


        if ($settings->auto_increment_assets == '1') {
            $temp_asset_tag = \DB::table('assets')
                ->where('physical', '=', '1')
                ->max('asset_tag');

            $asset_tag_digits = preg_replace('/\D/', '', $temp_asset_tag);
            $asset_tag = preg_replace('/^0*/', '', $asset_tag_digits);

            if ($settings->zerofill_count > 0) {
                return $settings->auto_increment_prefix.Asset::zerofill($settings->next_auto_tag_base, $settings->zerofill_count);
            }
            return $settings->auto_increment_prefix.$settings->next_auto_tag_base;
        } else {
            return false;
        }
    }
```

Is it just my lack of PHP knowledge, or are the variables `$temp_asset_tag`, `$asset_tag_digits`, `$asset_tag` never used anywhere else and those three assignments could be removed?

**Reply by Viiviv** [reply]
Thank you so much. This solved my issue.
